golang/trace/zipkin_tracereporter.go

```diff
@@ -116,18 +120,75 @@ func buildZipkinSpan(span tc.Span, annotations []tc.Annotation, binaryAnnotation
 		ServiceName: targetEndpoint.ServiceName,
 	}
 
-	// TODO Add BinaryAnnotations
+	tBinaryAnnotations, err := buildBinaryAnnotations(binaryAnnotations)
+	if err != nil {
+		return nil, err
+	}
 	thriftSpan := tcollector.Span{
-		TraceId:     uint64ToBytes(span.TraceID()),
-		Host:        &host,
-		Name:        targetEndpoint.Operation,
-		Id:          uint64ToBytes(span.SpanID()),
-		ParentId:    uint64ToBytes(span.ParentID()),
-		Annotations: buildZipkinAnnotations(annotations),
-		Debug:       false,
+		TraceId:           uint64ToBytes(span.TraceID()),
+		Host:              &host,
+		Name:              targetEndpoint.Operation,
+		Id:                uint64ToBytes(span.SpanID()),
+		ParentId:          uint64ToBytes(span.ParentID()),
+		Annotations:       buildZipkinAnnotations(annotations),
+		BinaryAnnotations: tBinaryAnnotations,
 	}
 
-	return &thriftSpan
+	return &thriftSpan, nil
+}
+
+func buildBinaryAnnotation(ann tc.BinaryAnnotation) (*tcollector.BinaryAnnotation, error) {
+	bann := &tcollector.BinaryAnnotation{Key: ann.Key}
+	switch v := ann.Value.(type) {
+	case bool:
+		bann.AnnotationType = tcollector.AnnotationType_BOOL
+		bann.BoolValue = &v
+	case int64:
+		bann.AnnotationType = tcollector.AnnotationType_I64
+		temp := v
+		bann.IntValue = &temp
+	case int32:
+		bann.AnnotationType = tcollector.AnnotationType_I32
+		temp := int64(v)
+		bann.IntValue = &temp
+	case int16:
+		bann.AnnotationType = tcollector.AnnotationType_I16
+		temp := int64(v)
+		bann.IntValue = &temp
+	case int:
+		bann.AnnotationType = tcollector.AnnotationType_I32
+		temp := int64(v)
+		bann.IntValue = &temp
+	case string:
+		bann.AnnotationType = tcollector.AnnotationType_STRING
+		bann.StringValue = &v
+	case []byte:
+		bann.AnnotationType = tcollector.AnnotationType_BYTES
+		bann.BytesValue = v
+	case float32:
+		bann.AnnotationType = tcollector.AnnotationType_DOUBLE
+		temp := float64(v)
+		bann.DoubleValue = &temp
+	case float64:
+		bann.AnnotationType = tcollector.AnnotationType_DOUBLE
+		temp := float64(v)
+		bann.DoubleValue = &temp
+	default:
+		return bann, errors.New("Unrecognized date type.")
```

Comment:
The error case should:
 - return nil
 - mention the type that couldn't be converted
 - use lower case error strings

```
return nil, fmt.Errorf("unrecognized data type: %T", v)
```